src/fw/drivers/sf32lb52/watchdog.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2025 Core Devices LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/watchdog.h"
+
+#include "util/bitset.h"
+#include "system/logging.h"
+#include <inttypes.h>
+#include "bf0_hal.h"
+
+#define WDT_RELOADER_TIMEOUT 3
+#define WDT_REBOOT_TIMEOUT 8
+
+static WDT_HandleTypeDef hwdt = {
+    .Instance = hwp_wdt1,
+};
+
+static int s_sys_freq = 0;
+static int s_irq_handle = 0;
+
+static HAL_StatusTypeDef wdt_set_timeout(WDT_HandleTypeDef *wdt, uint32_t reload_timeout) {
+  HAL_StatusTypeDef status;
+  wdt->Init.Reload = s_sys_freq * reload_timeout;
+  wdt->Init.Reload2 = s_sys_freq * WDT_REBOOT_TIMEOUT;
+
+  status = HAL_WDT_Init(wdt);
+  PBL_ASSERTN(status == HAL_OK);
+
+  __HAL_SYSCFG_Enable_WDT_REBOOT(1);
+  return HAL_OK;
+}
+
+void wdt_reconfig(void) {
+  HAL_WDT_Refresh(&hwdt);
+  __HAL_WDT_STOP(&hwdt);
```

Comment:
Does this `__HAL_WDT_STOP` stop the wdt.

Just an idear, like feed watchdog in this interrupt, and if system is stuck, the second step will reset system.